quic/s2n-quic-platform/src/io/turmoil.rs

```diff
@@ -0,0 +1,467 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
```

Comment:
This seems like a great starting point, but it would be great to fill the gaps so we just have tokio io and we can swap `turmoil::net` for `tokio::net` and it just works.

Reply:
The trickiest part (and why I didn't end up using the tokio provider) is we use some `libc` syscalls with raw FDs that you wouldn't be able to intercept. But if tokio had an API that reported to us if it was inside of a simulation or not, we could avoid those paths at runtime and have it Just Work™ for applications.